### PR TITLE
chore(web-components): Add playwright testing for v3 web components

### LINKED ISSUE #28078: Add unit test tooling to Fluent UI Web Components v3
Add unit test tooling to Fluent UI Web Components v3
- check what Fast Foundation does, see what we should and could cover
- can we do "50% of API coverage" bar?
- think about conformance tests

Based on async discussions, we currently incline to using Playwright over Jest (ESM support).

### PR DESCRIPTION
**Pull request by chrisdholt**
## Previous Behavior
No tests

## New Behavior
Tests!

## Related Issue(s)

- Closes #28078

### CODE REVIEW

**Comment by chrisdholt** [comment]
@Hotell Would love your help figuring out where/when to install playwright as part of the build. I expect this to fail since w/o a global playwright installation the chrome binaries won't be available.

File: packages/web-components/package.json [hunk 1]
```diff
@@ -140,43 +140,13 @@
...
-    "karma-coverage-istanbul-reporter": "3.0.3",
-    "karma-firefox-launcher": "^1.3.0",
-    "karma-mocha": "2.0.1",
-    "karma-mocha-reporter": "2.2.5",
-    "karma-source-map-support": "1.4.0",
-    "karma-sourcemap-loader": "0.3.8",
-    "karma-webpack": "5.0.0",
-    "mocha": "7.2.0",
     "@microsoft/api-extractor": "7.31.2",
+    "@playwright/test": "~1.28.1",
```
Comment (Hotell) [inline packages/web-components/package.json:147]: we need to move this to monorepo root package.json to follow single version policy. besides that the config etc can stay as is and we should be good to go @chrisdholt

File: packages/web-components/playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  use: {
+    baseURL: 'http://localhost:6006/iframe.html',
+    viewport: {
+      height: 1280,
+      width: 720,
+    },
+  },
+  webServer: {
+    // double-quotes are required for Windows
+    command: `node -e "import('express').then(({ default: e }) => e().use(e.static('./storybook-static')).listen(6006))"`,
```
Comment (miroslavstastny) [inline packages/web-components/playwright.config.ts:19]: What builds the storybook to `./storybook-static`?

File: packages/web-components/src/helpers.tests.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import qs from 'qs';
+
+/**
+ * Returns a formatted URL for a given Storybook fixture.
+ *
+ * @param id - the Storybook fixture ID
+ * @param args - Story args
+ * @returns - the local URL for the Storybook fixture iframe
+ */
+export function fixtureURL(id: string = 'debug--blank', args?: Record<string, any>): string {
```
Comment (miroslavstastny) [inline packages/web-components/src/helpers.tests.ts:10]: When I try to debug the test, the storybook renders with an error:
image

(The `radio` story is still rendered under the error).

File: packages/web-components/package.json (hunk 1 above)
Comment (chrisdholt) [reply packages/web-components/package.json]: @Hotell updated, thanks!

File: packages/web-components/playwright.config.ts (hunk 2 above)
Comment (chrisdholt) [reply packages/web-components/playwright.config.ts:19]: Locally you'd run the storybook server and then run `yarn test`. I added a note in the package readme about this. In the case of CI, it'll run the server itself, but locally we reuse it.

File: packages/web-components/src/helpers.tests.ts (hunk 3 above)
Comment (chrisdholt) [reply packages/web-components/src/helpers.tests.ts:10]: Interesting - repro steps?

image

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

File: packages/web-components/src/helpers.tests.ts (hunk 3 above)
Comment (miroslavstastny) [reply packages/web-components/src/helpers.tests.ts:10]: `yarn playwright test --headed --debug`

imo the fixtureURL should be `components-radio--radio`

Comment (chrisdholt) [reply packages/web-components/src/helpers.tests.ts:10]: Thanks for this, I'll look now

File: packages/web-components/src/radio/radio.spec.ts [hunk 4]
```diff
@@ -0,0 +1,290 @@
+import { expect, test } from "@playwright/test";
+import type { Locator, Page } from "@playwright/test";
+import { fixtureURL } from "../helpers.tests.js";
+import type { Radio } from "./radio.js";
+
+test.describe("Radio", () => {
```
Comment (miroslavstastny) [inline packages/web-components/src/radio/radio.spec.ts:6]: Formatting: single quotes. Have you run prettier on this?
I see `*.spec.ts` excluded in .prettierignore:
https://github.com/microsoft/fluentui/blob/5745de0ab6ab860f905c33d4430b857b8a72f27f/.prettierignore#L38-L39

Do you recall the reason? Can we re-enable it? Separate PR would be fine.

Comment (chrisdholt) [reply packages/web-components/src/radio/radio.spec.ts:6]: Yeah, I'll enable it - I think that's outdated. I think perhaps one reason was due to the fact it was messing with spacing in output for HTML so we'll want to keep an eye on it.

File: packages/web-components/src/helpers.tests.ts (hunk 3 above)
Comment (chrisdholt) [reply packages/web-components/src/helpers.tests.ts:10]: @miroslavstastny resolved!

File: packages/web-components/src/radio/radio.spec.ts (hunk 4 above)
Comment (chrisdholt) [reply packages/web-components/src/radio/radio.spec.ts]: Updated!

**Review by miroslavstastny — APPROVED** [review]

**Comment by JustSlone** [comment]
@TristanWatanabe @ling1726 we need sign off from fluent ui react build now that Martin's out, can you review this?

**Review by TristanWatanabe — APPROVED** [review]

**Pull request merged**